src/main/java/apoc/result/ConstraintRelationshipInfo.java

```diff
@@ -0,0 +1,101 @@
+package apoc.result;
+
+import org.neo4j.collection.primitive.PrimitiveIntIterator;
+import org.neo4j.collection.primitive.PrimitiveLongIterator;
+import org.neo4j.cursor.Cursor;
+import org.neo4j.kernel.api.ReadOperations;
+import org.neo4j.kernel.api.StatementTokenNameLookup;
+import org.neo4j.kernel.api.constraints.RelationshipPropertyConstraint;
+import org.neo4j.kernel.api.exceptions.EntityNotFoundException;
+import org.neo4j.kernel.api.exceptions.LabelNotFoundKernelException;
+import org.neo4j.kernel.api.exceptions.RelationshipTypeIdNotFoundKernelException;
+import org.neo4j.storageengine.api.RelationshipItem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by alberto.delazzari on 04/07/17.
+ */
+public class ConstraintRelationshipInfo {
+
+    public final String name;
+
+    public final List<String> startLabel;
+
+    public final String type;
+
+    public final List<String> endLabel;
+
+    public final List<String> properties;
+
+    public final String status;
+
+    public ConstraintRelationshipInfo(ReadOperations readOperations, RelationshipPropertyConstraint propertyConstraint) throws Exception {
+        StatementTokenNameLookup lookup = new StatementTokenNameLookup(readOperations);
+
+        name = propertyConstraint.userDescription(lookup);
+        properties = Collections.singletonList(lookup.propertyKeyGetName(propertyConstraint.propertyKey()));
+        status = "";
+        type = lookup.relationshipTypeGetName(propertyConstraint.relationshipType());
+
+        LabelResult labelResult = getNodeLabelsFromRelationshipTypeId(readOperations, propertyConstraint.relationshipType());
+
+        startLabel = labelResult.startLabels;
+        endLabel = labelResult.endLabels;
+    }
+
+    private LabelResult getNodeLabelsFromRelationshipTypeId(ReadOperations readOperations, int relationshipTypeId)
```

Comment:
there is schema.getIndexState http://neo4j.com/docs/java-reference/current/javadocs/org/neo4j/graphdb/schema/Schema.html#getIndexState-org.neo4j.graphdb.schema.IndexDefinition-